### PR TITLE
[FIX] fields: cache consistency for x2many fields

### PR DESCRIPTION
**Pull request by rco-odoo**
This aims at fixing cache inconsistencies for x2many fields with inactive records.
The fix consists in having only one value in cache, instead of two (corresponding to `active_test=True` and `active_test=False`).

### CODE REVIEW

File: odoo/addons/test_new_api/tests/test_new_fields.py [hunk 1]
```diff
@@ -1539,6 +1539,50 @@ def test_11_ondelete_many2many(self):
...
+        self.assertEqual(parent.with_context(active_test=True).children_ids, act_children)
+        self.assertEqual(parent.with_context(active_test=False).children_ids, all_children)
+
+        # create with active_test=False in context
+        child3, child4 = Model.with_context(active_test=False).create([
+            {'parent_id': parent.id, 'active': True},
+            {'parent_id': parent.id, 'active': False},
+        ])
+        act_children = child1 + child3
+        all_children = child1 + child2 + child3 + child4
```
Comment (Elkasitu) [inline odoo/addons/test_new_api/tests/test_new_fields.py]: `+=`

**Comment by beledouxdenis** [comment]
Nice & clean I like it!

**Comment by qsm-odoo** [comment]
@rco-odoo Installing themes does not work yet when we try to *switch* themes, traceback.

After debugging:
this line https://github.com/odoo/odoo/blob/saas-12.5/odoo/fields.py#L1062 triggers a cache miss even though I go through that line https://github.com/odoo/odoo/blob/saas-12.5/odoo/addons/base/models/ir_module.py#L946 with the same record and I put a value in there just before.

cc @seb-odoo

**Comment by seb-odoo** [comment]
The problem seems to be that `_compute_field_value` in `models.py` is writing on the cache of another environment than the one present on `compute_value` in `fields.py`.

Probably because of the `sudo` just above in `compute_value`.

**Comment by seb-odoo** [comment]
Issue highlighted in https://github.com/odoo/odoo/pull/37735

The warning will be logged when trying to change the theme of a website.

**Pull request closed without merging**